flows/actions/send_msg.go

```diff
@@ -142,40 +142,34 @@ func getTemplatingMsg(action *SendMsgAction, run flows.Run, urn urns.URN, channe
 	}
 	evaluatedText := templateTranslation.Substitute(evaluatedVariables)
 
-	evaluatedParams := make(map[string][]flows.TemplateParam)
-	for compKey, compParams := range action.Templating.Params {
-		compVariables := make([]flows.TemplateParam, len(compParams))
-		for i, templateParam := range compParams {
-			var paramValue string
-			var err error
-			if strings.HasPrefix(compKey, "button.") {
+	evaluatedParams := make(map[string]flows.ComponentVariables)
+	for compKey, compVars := range action.Templating.Params {
 
-				qrIndex, err := strconv.Atoi(strings.TrimPrefix(compKey, "button."))
-				if err != nil {
-					logEvent(events.NewError(err))
-				}
-				paramValue = evaluatedQuickReplies[qrIndex]
-			} else if templateParam.Type() != "text" {
-				paramValue = ""
-				for _, att := range evaluatedAttachments {
-					attType := strings.Split(att.ContentType(), "/")[0]
-					if templateParam.Type() == attType {
-						paramValue = att.URL()
-						break
-					}
-				}
-			} else {
-				localizedParamVariables, _ := run.GetTextArray(uuids.UUID(templateParam.UUID()), "value", []string{templateParam.Value()}, nil)
-				paramValue, err = run.EvaluateTemplate(localizedParamVariables[0])
+		var evaluatedComponentVariables []string
+		if strings.HasPrefix(compKey, "button.") {
+			qrIndex, err := strconv.Atoi(strings.TrimPrefix(compKey, "button."))
+			if err != nil {
+				logEvent(events.NewError(err))
+			}
+			paramValue := evaluatedQuickReplies[qrIndex]
+			evaluatedComponentVariables = []string{paramValue}
+
```

Comment:
I can use the quick replies for the node for buttons however no easyway to detect the header expects the media file so we use the attachments

I think that will require the user to always upload the file for the attchments and the template field